docs/source/getting-started.md

```diff
@@ -0,0 +1,194 @@
+
+#### Runtime Libraries
+ExecuTorch provides a native C++ runtime API, as well as platform and language-specific bindings for Python, Android, and iOS.
+
+##### Python
+Python runtime bindings are provided as part of the executorch pip package.
+
+##### Android
+ExecuTorch Android Java libraries are available as a pre-built library. To add the library to your app, download the AAR, and add it to the gradle build rule. TODO Replace with Maven/Gradle package management when available.
+```
+mkdir -p app/libs
+curl https://ossci-android.s3.amazonaws.com/executorch/release/executorch-241002/executorch.aar -o app/libs/executorch.aar
+```
+And in gradle,
+```
+# app/build.gradle.kts
+dependencies {
+    implementation(files("libs/executorch.aar"))
+}
+```
+
+##### iOS
+The iOS runtime library is provided as a collection of .xcframework targets and are made available as a Swift PM package. To get started with Xcode, go to File > Add Package Dependencies. Paste the URL of the ExecuTorch repo into the search bar and select it. Make sure to change the branch name to the desired ExecuTorch version in format “swiftpm-”, (e.g. “swiftpm-0.4.0”), or a branch name in format “swiftpm-.<year_month_date>” (e.g. “swiftpm-0.4.0-20241201”) for a nightly build on a specific date.
+
+The ExecuTorch dependency can also be added to the package file manually. See [Using ExecuTorch on iOS](/TODO.md) for more information.
+
+##### C++
+
+CMake is the preferred build system for the ExecuTorch C++ runtime. To use with CMake, clone the ExecuTorch repository as a subdirectory of your project, and use CMake's `add_subdirectory("executorch")` to include the dependency. The `executorch` target, as well as kernel and backend targets will be made available to link against. The runtime can also be built standalone to support diverse toolchains. See [Using ExecuTorch with C++](/TODO.md) for a detailed description of build integration, targets, and cross compilation.
+
+```
+git clone -b release/0.5 https://github.com/pytorch/executorch.git
+```
+```python
+# CMakeLists.txt
+add_subdirectory("executorch")
+...
+target_link_libraries(
+  my_target
+  PRIVATE executorch
+          executorch_module_static
+          executorch_tensor
+          optimized_native_cpu_ops_lib
+          xnnpack_backend)
+```
+
+<hr/>
+
+## Preparing the Model
+Exporting is the process of taking a PyTorch model and converting it to the .pte file format used by the ExecuTorch runtime. This is done using Python APIs. PTE files for common models can be found on HuggingFace (TODO add link).
+
+### Requirements
+- A PyTorch model.
+- Example model inputs, typically as PyTorch tensors. You should be able to successfully run the PyTorch model with these inputs.
+- One or more target hardware backends.
+
+### Selecting a Backend
+ExecuTorch provides hardware backends for a wide variety of hardware. The most commonly used backends are XNNPACK, for ARM and x86 CPU, CoreML (for iOS), and Vulkan (for Android GPUs).
```

Comment:
Mention Qualcomm for Android NPU

Reply:
Added. I do think that it is very hard to use NPU delegates right now, in part because we don't expose any sane way to detect if the user is running supported hardware, and they need to ship a fully separate PTE. Hopefully backend capabilities + program data separation will help alleviate this. Until then, I think it's a hard sell to get anyone to do the extra work unless they really want every bit of performance. Internally, I've had a hard time getting people to sign up to support multiple delegates because of the amount of work needed to export another PTE + test it + write runtime code. I'd like to focus on this once we get the core CPU + CoreML experience tightened up.